### PR TITLE
Updating DFOCC: Adding DF-CCSD(T) analytic grads for RHF ref.

### PR DESCRIPTION
**Pull request by bozkaya**
## Description
Updating DFOCC module with new features

## Todos
Notable points that this PR has either accomplished or will accomplish.
* **User-Facing for Release Notes**
  - [ 1 ] DF-CCSD(T) analytic grads are implemented
  - [ 2 ] EKT is implemented for DFOCC module

## Questions
- [x] Question1

## Status
- [x] Ready to go

### CODE REVIEW

File: psi4/driver/procrouting/proc.py [hunk 1]
```diff
@@ -873,7 +873,11 @@ def select_ccsd_t__gradient(name, **kwargs):
     # Considering only ccenergy
 
     func = None
-    if reference in ['RHF', 'UHF']:
+    if reference in ['RHF']:
+        if mtd_type == 'DF':
+            if module in ['', 'OCC']:
+                func = run_dfocc_gradient
+    elif reference == 'UHF':
```
Comment (loriab) [inline psi4/driver/procrouting/proc.py]: This change loses the ability to run RHF CONV gradients via ccenergy. Easy to edit it back.

File: psi4/driver/procrouting/proc.py [hunk 2]
```diff
@@ -1611,10 +1615,13 @@ def run_dfocc_gradient(name, **kwargs):
...
         core.set_local_option('DFOCC', 'WFN_TYPE', 'DF-CCSD')
         core.set_local_option('DFOCC', 'CC_LAMBDA', 'TRUE')
+    elif name in ['ccsd(t)']:
+        core.set_local_option('DFOCC', 'WFN_TYPE', 'DF-CCSD(T)')
+        core.set_local_option('DFOCC', 'CC_LAMBDA', 'TRUE')
     else:
         raise ValidationError('Unidentified method %s' % (name))
 
-    if name in ['mp2', 'mp2.5', 'mp3', 'lccd', 'ccd', 'ccsd']:
+    if name in ['mp2', 'mp2.5', 'mp3', 'lccd', 'ccd', 'ccsd', 'ccsd(t)', 'ccsd(at)']:
```
Comment (loriab) [inline psi4/driver/procrouting/proc.py]: I don't think `ccsd(at)` can be reached here at present? Not in a `select_` fn or in the options setting directly above.

File: psi4/src/psi4/dfocc/ccsd_tpdm.cc [hunk 3]
```diff
@@ -31,60 +31,59 @@
...
     G->axpy(V, 2.0);
     V.reset();
-    // G_ij^Q += P+(ij) V_ij^Q
+    // G_ij^Q += P+(ij) V_ij^Q 
     V = SharedTensor2d(new Tensor2d("V (Q|IJ)", nQ, naoccA, naoccA));
     V->read(psio_, PSIF_DFOCC_AMPS);
     G->axpy(V, 1.0);
     V.reset();
-    // G_ij^Q += P+(ij) Vt_ij^Q
+    // G_ij^Q += P+(ij) Vt_ij^Q
```
Comment (loriab) [inline psi4/src/psi4/dfocc/ccsd_tpdm.cc]: There's a nifty format file to drive `clang-format` now. It'll get rid of unnecessary whitespace additions, I hope. And uneven indentation

File: psi4/src/psi4/dfocc/mp3_W_intr.cc [hunk 4]
```diff
@@ -31,6 +31,8 @@
 #include "dfocc.h"
 
 using namespace psi;
+using namespace std;
```
Comment (loriab) [inline psi4/src/psi4/dfocc/mp3_W_intr.cc]: We've purged `using namespace std` from the rest of the code for clarity. So please go ahead and `std::` anything that needs it.

File: psi4/driver/procrouting/proc.py (hunk 1 above)
Comment (loriab) [reply psi4/driver/procrouting/proc.py]: Actually, we already patched that up once [here](https://github.com/loriab/privUgurPsi4/commit/d4bbc369fa8a34871bd3e8950ffc4267354525d8). Maybe see if any other improvements on that branch didn't make it to this PR?

File: psi4/src/psi4/dfocc/ccsd_tpdm.cc (hunk 3 above)
Comment (bozkaya) [reply psi4/src/psi4/dfocc/ccsd_tpdm.cc]: The following revisions were made based on Lori's comments:

* DFOCC module I run clang-format
* using namespace std; is deleted from mp3_W_intr.cc file
* proc is revised, now we can run analytic gradients with both ccenergy and dfocc 

@loriab

Comment (loriab) [reply psi4/src/psi4/dfocc/ccsd_tpdm.cc]: Thanks, @bozkaya, but I don't think the clang-format worked. I tried it locally (`clang-format -i file.cc` on a psi4 repo with `.clang-format` in top-level) on one of your files, and it did get rid of the spaces at end-of-line and uneven indentation. So maybe give it another try.

**Comment by bozkaya** [comment]
Hi Lori,

Now, the dfocc source code is properly clang-formatted. I also update the docs. @loriab
